New retry predicate for Global Usage Limits (429 with a domain of "…

kcibul opened the pull request:
- New retry predicate for Global Usage Limits (429 with a domain of "global")
- Applied to service account creation, key generation, and removal (have seen this rate limit in performance testing w/ > 250 SAs being created)

**PR checklist**
- [x] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [x] Bump the version in `project/Settings.scala` `createVersion()`
- [x] Update `CHANGELOG.md` for those libraries
- [x] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [x] Replace the appropriate version hashes in `README.md` and the `CHANGELOG.md` for any libs you changed with `TRAVIS-REPLACE-ME` to auto-update the version string
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

kcibul committed:
- Added new predicate for Global Usage Limits (429 with a domain of "global")
- Applied to service account creation, key generation, and removal (have seen this rate limit in performance testing w/ > 250 SAs being created)

helgridly commented on README.md:
```diff
@@ -39,11 +39,11 @@ Latest SBT dependency: `"org.broadinstitute.dsde.workbench" %% "workbench-metric
 Contains utility functions for talking to Google APIs and DAOs for Google PubSub, Google Directory, Google IAM, and Google BigQuery. 
 
-Latest SBT dependency: `"org.broadinstitute.dsde.workbench" %% "workbench-google" % "0.20-a9f29eb"`
+Latest SBT dependency: `"org.broadinstitute.dsde.workbench" %% "workbench-google" % "0.21-e4cdd00"`
```
this (and friends) should be `0.21-TRAVIS-REPLACE-ME`. There's a script that runs at the end of the Travis build step that uses the commit hash at the top of develop, updates the string, and pushes the lib with the commit hash in it for you

kcibul committed:
- Added new predicate for Global Usage Limits (429 with a domain of "global")
- Applied to service account creation, key generation, and removal (have seen this rate limit in performance testing w/ > 250 SAs being created)

kcibul committed:
- added tests

dvoet commented on project/Settings.scala:
```diff
@@ -129,7 +129,7 @@ object Settings {
     name := "workbench-google",
     libraryDependencies ++= googleDependencies,
-    version := createVersion("0.20"),
+    version := createVersion("0.21"),
```
this looks backwards compatible so no need to increment the version

kcibul committed:
- added tests
